packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -345,6 +345,16 @@ export const coreDefinitions: {[coreSettingName: string]: SettingsDefinition} =
           type: SettingsType.BOOLEAN,
           default: null,
         },
+        httpProxy: {
+          description: `URL of the http proxy that must be used for outgoing http requests`,
+          type: SettingsType.STRING,
+          default: process.env.HTTP_PROXY || process.env.http_proxy || null,
+        },
+        httpsProxy: {
+          description: `URL of the http proxy that must be used for outgoing https requests`,
+          type: SettingsType.STRING,
+          default: process.env.HTTPS_PROXY || process.env.https_proxy || null,
```

Comment:
Shouldn't these two default to `null`?
Right now, a configuration like

```yaml
httpsProxy: 'https://proxy.local'

networkSettings:
  custom-ca.local:
    caFilePath: /etc/custom-ca.cert
```

will make requests to `custom-ca.local` use the proxy configured via `HTTPS_PROXY`—if present in the environment—while all other domains will use the configured `httpsProxy`. If `HTTPS_PROXY` is not set in the environment, the configured `httpsProxy` will be used for all requests.

Reply:
Indeed they should, cheers